src/test/test_runner_host.py

```diff
@@ -197,6 +197,147 @@ def test_docker_basic_run(self):
+                "runs": ["echo hello"],
+                "id": "kontainer_one",
+            },
+            default_box=True,
+        )
+        cid = pu.sanitized_name(step.id, config.wid)
+        with PodmanRunner(init_podman_client=True, config=config) as pr:
+            c = pr._find_container(cid)
+            self.assertEqual(c, None)
+        with PodmanRunner(init_podman_client=True, config=config) as pr:
+            container = pr._create_container(cid, step)
+            c = pr._find_container(cid)
+            self.assertEqual(c, container)
+            cmd = ["podman", "container", "rm", "-f", cid]
+            HostRunner._exec_cmd(cmd, logging=False)
+
+    @unittest.skipIf(os.environ.get("ENGINE", "docker") != "podman", "ENGINE != podman")
+    def test_create_container(self):
+        config = ConfigLoader.load()
+        step = Box(
+            {
+                "uses": "docker://alpine:3.9",
+                "runs": ["echo hello"],
+                "id": "kontainer_one",
+            },
+            default_box=True,
+        )
+        cid = pu.sanitized_name(step.id, config.wid)
+        with PodmanRunner(init_podman_client=True, config=config) as pr:
+            c = pr._create_container(cid, step)
+            c_status_cmd = [
+                "podman",
+                "container",
+                "inspect",
+                "-f",
+                str("{{.State.Status}}"),
+                c,
+            ]
+            __, _, c_status = HostRunner._exec_cmd(c_status_cmd, logging=False)
+            self.assertEqual(c_status, "configured\n")
+            cmd = ["podman", "container", "rm", c]
+            HostRunner._exec_cmd(cmd, logging=False)
+        step = Box(
+            {
+                "uses": "docker://alpine:3.9",
+                "runs": ["echo", "hello_world"],
+                "id": "KoNtAiNeR tWo",
+            },
+            default_box=True,
+        )
+        cid = pu.sanitized_name(step.id, config.wid)
+        with PodmanRunner(init_podman_client=True, config=config) as pr:
+            c = pr._create_container(cid, step)
+            c_status_cmd = [
+                "podman",
+                "container",
+                "inspect",
+                "-f",
+                str("{{.State.Status}}"),
+                c,
+            ]
+            __, _, c_status = HostRunner._exec_cmd(c_status_cmd, logging=False)
+            self.assertEqual(c_status, "configured\n")
+            cmd = ["podman", "container", "rm", c]
+            HostRunner._exec_cmd(cmd, logging=False)
+
+    @unittest.skipIf(os.environ.get("ENGINE", "docker") != "podman", "ENGINE != podman")
+    def test_podman_basic_run(self):
+        repo = self.mk_repo()
+        conf = ConfigLoader.load(engine_name="podman", workspace_dir=repo.working_dir)
+
+        with WorkflowRunner(conf) as r:
+            # wf_data = {"steps": [{"uses": "popperized/bin/sh@master", "args": ["ls"],}]}
```

Comment:
Yes it is.

Reply:
hmm, it'd be great if we could to find the root cause, as this shouldn't fail.

Also, it'd be good to include a workflow that fails so that we ensure that the runners return non-zero exit code. This would catch the issue with 2.0.3 we faced.

Reply:
I think I've found the issue. Let me give a detailed explanation.

```['podman', 'create', '--name', 'popper_1_483f7d07', '-v', '/tmp/tmpn9poi6r_:/workspace:Z', '-e', 'GIT_COMMIT=f81810d4e55a4586d7143760722a522f51e08570', '-e', 'GIT_BRANCH=master', '-e', 'GIT_SHA_SHORT=f81810d', '-e', 'GIT_REMOTE_ORIGIN_URL=https://github.com/my/repo', '-d', '-w', '/workspace', 'popperized/bin:master', 'ls']
```
When the above command runs, Podman tries to pull the image `popperized/bin:master` from the docker repository as evidenced by the results shown below.

```', 'to', 'retrieve', 'auth', 'token:', 'invalid', 'username/password:', 'unauthorized:', 'Please', 'login', 'to', 'the', 'Red', 'Hat', 'Registry', 'using', 'your', 'Customer', 'Portal', 'credentials.', 'Further', 'instructions', 'can', 'be', 'found', 'here:', 'https://access.redhat.com/RegistryAuthentication', 'Trying', 'to', 'pull', 'docker.io/popperized/bin:master...', 'Getting', 'image', 'source', 'signatures', 'Copying', 'blob', 'sha256:ed42820a5a6a336e9ba51dd866d035f9500fdc4fbb07acd2bc40283820769778', 'Copying', 'blob', 'sha256:55e542d1686ed5e6a39f48ca2ee55c47d9cc1513a97f8783dad60e5c7bfbd8ae', 'Copying', 'blob', 'sha256:412bf51e209fa1182aa4d0b61fb144a9dc8ab885b614fd29327773c6baa4a39d', 'Copying', 'config', 'sha256:036ccbea75797737bd1824eeb8372509f75b2575a7b96245c1fea7503881a874', 'Writing', 'manifest', 'to', 'image', 'destination', 'Storing', 'signatures', 'e63945985d63231b26b307054c40cd9177275023226f89876718270b22e80ba0']
```

The [line](https://github.com/edeediong/popper/blob/ddc4e2e5538b49d8b6becca26d499c8874264c9c/src/popper/runner_host.py#L379) here tells the runner to pick the first value which in this case is `Trying/n` and not the container id. Seems when pulling from the `popperized/bin:master`, there's a delay in the VM. 

However, I found a workaround: `container[0]` will be changed to `container[-1]` to accommodate this issue.

Reply:
I've written a workflow which [fails](https://travis-ci.com/github/getpopper/popper/jobs/367912849). I just have to specify the version of Podman in the Travis script.

Reply:
Update: There seems to be no way to specify version of podman to install on travis because it doesn't exist in the openSuse [repository](https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/testing/xUbuntu_18.04/). The test presently failing @ivotron is the test that returns non-zero code. It works on v1.x but not on 2.x at least not until the code fixes are pushed to upstream